lib/rules/padding-line-between-tags.js

```diff
@@ -0,0 +1,189 @@
+      const paddingText = context.getSourceCode().text.slice(start, end)
+      const textBetween = splitLines(paddingText)
+      let newTextBetween = `\n${textBetween.pop()}`
+      for (let i = textBetween.length - 1; i >= 0; i--) {
+        if (!/^\s*$/.test(textBetween[i])) {
+          newTextBetween = `${i === 0 ? '' : '\n'}${
+            textBetween[i]
+          }${newTextBetween}`
+        }
+      }
+      return fixer.replaceTextRange([start, end], `${newTextBetween}`)
+    }
+  })
+}
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+/**
+ * @param {RuleContext} context
+ */
+function checkNewline(context) {
+  /** @type {Array<{blankLine: "always" | "never", prev: string, next: string}>} */
+  const configureList = context.options[0] || [
+    { blankLine: 'always', prev: '*', next: '*' }
+  ]
+
+  /**
+   * @param {VElement} block
+   */
+  return (block) => {
+    if (!block.parent.parent) {
+      return
+    }
+
+    const endTag = block.endTag || block.startTag
+    const lowerSiblings = block.parent.children
+      .filter(
+        (element) =>
+          element.type === 'VElement' && element.range !== block.range
+      )
+      .filter((sibling) => sibling.range[0] - endTag.range[1] >= 0)
+
+    if (lowerSiblings.length === 0) {
+      return
+    }
+
+    const closestSibling = /** @type {VElement} */ (lowerSiblings[0])
+
+    for (let i = configureList.length - 1; i >= 0; --i) {
+      const configure = configureList[i]
+      const matched =
+        (configure.prev === '*' || block.name === configure.prev) &&
+        (configure.next === '*' || closestSibling.name === configure.next)
+
+      if (matched) {
+        const lineDifference =
+          closestSibling.loc.start.line - endTag.loc.end.line
+        if (configure.blankLine === 'always') {
+          if (lineDifference === 1) {
+            insertNewLine(context, block, closestSibling)
+          } else if (lineDifference === 0) {
+            context.report({
+              messageId: 'always',
+              loc: closestSibling.loc,
+              // @ts-ignore
+              fix(fixer) {
+                const lastSpaces = /** @type {RegExpExecArray} */ (
+                  /^\s*/.exec(
+                    context.getSourceCode().lines[endTag.loc.start.line - 1]
+                  )
+                )[0]
+
+                return fixer.insertTextAfter(endTag, `\n\n${lastSpaces}`)
+              }
+            })
+          }
+        } else {
+          if (lineDifference > 1) {
+            let hasOnlyTextBetween = false
+            for (
+              let i = endTag.loc.start.line;
+              i < closestSibling.loc.start.line - 1;
+              i++
+            ) {
+              hasOnlyTextBetween = !/^\s*$/.test(
+                context.getSourceCode().lines[i]
+              )
+            }
```

Comment:
I think it's the same as checking only the last line.
Or does it require additional processing?

Reply:
Hi, sorry, this was a mistake from programming while tired :)
Thank you for finding this